proposals/NNNN-android-manager-apis.md

```diff
@@ -0,0 +1,689 @@
+            void onError(String error);
+        }
+    }
+}
+```
+
+#### VideoStreamingManager
+
+The Android library has a private manager that currently manages video streaming through a `SdlRemoteDisplay` class. This manager intends to be developer facing and handle both cases of remote display and give the developer the option to stream bytes of their encoding through the video service. 
+
+```java
+@TargetApi(19)
+public class VideoStreamingManager extends BaseSubManager implements ISdlServiceListener {
+
+    public VideoStreamingManager(Context context,ISdl internalInterface){}
+
+    //Library handled video stream display
+    public void startRemoteDisplayStream(final Class<? extends SdlRemoteDisplay> remoteDisplay, final VideoStreamingParameters parameters, final boolean encrypted){}
+    
+    //Method to receive an interface to write raw video 
+    public IVideoStreamListener startVideoStream(VideoStreamingParameters parameters, boolean encrypted){}
+    
+    private void startVideoStreaming(VideoStreamingParameters parameters, boolean encrypted){}
+
+    public void stopStreaming(){}
+
+	@Override
+    public void dispose(){}
+
+    @Override
+    public void onServiceStarted(SdlSession session, SessionType type, boolean isEncrypted) {}
+
+    @Override
+    public void onServiceEnded(SdlSession session, SessionType type) {}
+
+    @Override
+    public void onServiceError(SdlSession session, SessionType type, String reason) {}
+}
+```
+
+#### FileManager
+
+The `FileManager` handles one of the most error prone SDL tasks, managing files uploads, status, and deletion. It is based on a few previously accepted and implemented proposal for the iOS library. This manager will contain a list of remote files that are retrieved on startup, and contains functions for the creating and deleting of files and artwork. It will also make use of our new `CompletionListener` for many of its functions.
+
+```java
+public class FileManager extends BaseSubManager {
+
+    FileManager(ISdl internalInterface, Context context) {}
+
+    // GETTERS
+
+    public ArrayList<String> getRemoteFileNames() {}
+
+    // DELETION
+
+    public void deleteRemoteFileWithName(String fileName, CompletionListener listener){}
+
+    public void deleteRemoteFilesWithNames(ArrayList<String> fileNames, CompletionListener listener){}
+
+    // UPLOAD FILES / ARTWORK
+
+    public void uploadFile(SDLFile file, CompletionListener listener){}
+
+    public void uploadFiles(ArrayList<SDLFile> files, CompletionListener listener){}
+    
+    // HELPER METHODS
+    
+    public static Uri resourceToUri(Context context, int resID)
+
+}
+```
+
+#### PermissionManager
+
+The `PermissionManager` allows the developer to easily query whether an individual permission is allowed or not. It also allows a listener to be added for a list of permissions so that if they are changed, the developer would then be notified. This manager closely mimics that of its iOS counterpart. 
```

Comment:
Can you include a link to the iOS Permission Manager proposal and/or implementation?